src/Microsoft.Extensions.OData.Migration/Formatters/Deserialization/ODataMigrationInputFormatter.cs

```diff
@@ -0,0 +1,238 @@
+﻿// ------------------------------------------------------------------------------
+// <copyright company="Microsoft Corporation">
+//     Copyright © Microsoft Corporation. All rights reserved.
+// </copyright>
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Extensions.OData.Migration.Formatters.Deserialization
+{
+    using Microsoft.AspNet.OData.Extensions;
+    using Microsoft.AspNet.OData.Formatter;
+    using Microsoft.AspNet.OData.Formatter.Deserialization;
+    using Microsoft.AspNet.OData.Routing;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Http.Headers;
+    using Microsoft.AspNetCore.Mvc.Formatters;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.OData.Migration.Formatters.Deserialization;
+    using Microsoft.Net.Http.Headers;
+    using Microsoft.OData;
+    using Microsoft.OData.Edm;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Input formatter that supports V3 conventions in request bodies
+    /// </summary>
+    public class ODataMigrationInputFormatter : ODataInputFormatter
+    {
+        public ODataMigrationInputFormatter(IEnumerable<ODataPayloadKind> payloadKinds)
+            : base(payloadKinds)
+        {
+            SupportedMediaTypes.Add(MediaTypeHeaderValue.Parse("application/json"));// investigate content-type →application/json;odata.metadata=minimal;odata.streaming=true;IEEE754Compatible=false;charset=utf-8
+            SupportedEncodings.Add(Encoding.UTF8);
+            SupportedEncodings.Add(Encoding.Unicode);
+        }
+
+        /// <summary>
+        /// Determine if incoming request is specifically OData v3; if not, then use the next InputFormatter
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public override bool CanRead(InputFormatterContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            if (context.HttpContext.Request.Headers.ContainsKey("dataserviceversion") ||
```

Comment:
Should case insensitive string comparison be used?

Reply:
The Migration middleware will be called first (tested) and should convert to lowercase.  Do you think that conversion should be totally case insensitive (e.g., dAtASeRviceVerSion should still be converted to dataserviceversion?)